packages/import-maps-generate/src/findPathToVersion.js

```diff
@@ -0,0 +1,87 @@
+import path from 'path';
+import fs from 'fs';
+import { findPackageJson } from './findPackageJson';
+
+async function findWorkspaceParentDeps(depName, packageJson, targetPath = process.cwd()) {
+  const parents = [];
+  if (packageJson.workspaces) {
+    // eslint-disable-next-line no-restricted-syntax
+    for (const globString of packageJson.workspaces) {
+      // eslint-disable-next-line no-await-in-loop
+      const packageJsonPaths = await findPackageJson(globString, targetPath);
+      packageJsonPaths.forEach(packageJsonPath => {
+        const packageJsonString = fs.readFileSync(packageJsonPath, 'utf-8');
+        const wsPackageJson = JSON.parse(packageJsonString);
+        if (wsPackageJson.dependencies && wsPackageJson.dependencies[depName]) {
+          parents.push(wsPackageJson.name);
+        }
+      });
+    }
+  }
+  return parents;
+}
+
+function findLocalParentDeps(depName, deps) {
+  const parentDeps = [];
+  Object.keys(deps).forEach(key => {
+    const dep = deps[key];
+    if (dep.dependencies && Object.keys(dep.dependencies).includes(depName)) {
+      const currentDepName = key.slice(0, key.lastIndexOf('@'));
+      parentDeps.push(currentDepName);
+    }
+  });
+  return parentDeps;
+}
+
+async function findAllParentDeps(depName, deps, packageJson, targetPath = process.cwd()) {
+  const parents = [
+    ...findLocalParentDeps(depName, deps),
+    ...(await findWorkspaceParentDeps(depName, packageJson, targetPath)),
+  ];
+  return parents;
+}
+
+function readJsonFile(filePath) {
+  const fileContent = fs.readFileSync(filePath, 'utf-8');
+  return JSON.parse(fileContent);
+}
+
+export async function findPathToVersion(
+  depName,
+  version,
+  deps,
+  packageJson,
+  targetPath = process.cwd(),
+) {
+  const rootVersionPath = require.resolve(`${depName}/package.json`, {
+    paths: [targetPath],
+  });
+  const { version: rootVersion } = readJsonFile(rootVersionPath);
+
+  if (rootVersion !== version) {
+    const parents = await findAllParentDeps(depName, deps, packageJson, targetPath);
+
+    for (let i = 0; i < parents.length; i += 1) {
+      const parent = parents[i];
+      const parentPath = require.resolve(parent, {
+        paths: [targetPath],
+      });
+      const subVersionPath = require.resolve(`${depName}/package.json`, {
```

Comment:
need to use `path.join` here

Reply:
are you sure? can't resolve handle pathes on its own?

Reply:
on windows it's a different path delimiter, right?

Reply:
jup but it can handle it... at least it work if I run it in windows 10 powershell
but I found another thing... I use path.join somewhere and that generates pathes with `\` in the "values" and they then get escaped so it's `\\` on windows..

long story short it creates "invalid" import maps on windows 😭

Reply:
so I am not sure should the import map we generate be normalized for linux? I guess? or should we normalize the pathes when we parse the import-map?

let's postpone this discussion for later... let's say linux pathes only for now 😢 

shall I add an info to the readme?

Reply:
or open an issue?

Reply:
added it to readme for now